### PR TITLE
Part 1 - refactor: use textContent instead of innerText

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@saschanaz sending quite a large refactor in parts for you.

File: src/core/algorithms.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
-Currently used only for adding 'assert' class to algorithm lists  
+Currently used only for adding 'assert' class to algorithm lists
```
Comment (saschanaz) [inline src/core/algorithms.js]: Maybe we should add `no-trailing-space` eslint rule.

Comment (marcoscaceres) [reply src/core/algorithms.js]: Yeah, would be good.

File: src/core/utils.js [hunk 2]
```diff
@@ -211,7 +211,7 @@ export function normalizePadding(text = "") {
   }
   doc.normalize();
   // use the first space as an indicator of how much to chop off the front
-  const firstSpace = doc.body.innerText
+  const firstSpace = doc.body.textContent
```
Comment (saschanaz) [inline src/core/utils.js]: I think we should keep this as it is... Or we just replace this bulky function with `reindent()` in `core/reindent`.

Comment (marcoscaceres) [reply src/core/utils.js]: I'd like to eventually use `reindent()` - that would be great. I need to try it and make sure it handles all the special cases that this one needed to deal with.

Comment (marcoscaceres) [reply src/core/utils.js]: In any case, I don't think there is any valid reason for this to be using `innerText`

Comment (saschanaz) [reply src/core/utils.js]: I was concerned about `innerText`s special behavior that converts `<br>` to `\n` and we are using the exact character here. Thinking more, yeah, I think it's okay. This is currently only for markdown and IDL strings that won't (normally) use any `<br>`.

Comment (marcoscaceres) [reply src/core/utils.js]: Understood. I didn't think of that, but I agree that it should be ok. The markdown is supposed to be processed as given, not as displayed.

**Pull request merged**